net/nginx/files/nginx.init

```diff
@@ -75,6 +75,40 @@ nginx_init() {
 	logger -t "nginx_init" -p "daemon.info" "using ${CONF} (the test is ok)"
 }
 
+add_mdns() {
+        local cfg="$1"
+        local port enabled mdns service
+
+        config_get enabled "$cfg" enabled
+        config_get mdns "$cfg" mdns
+        config_get port "$cfg" listen
+
+        port=$(echo "$port" | head -n1 | awk '{print $1;}')
+
+        if [ "${mdns}" == "0" ] || [ "${mdns}" == "false" ]; then
+                return
+        fi
+
+        if [ "${enabled}" != "0" ] && [ "${enabled}" != "false" ] && [ -n "${port}" ]; then
+		[ "${port}" == "80" ] && service="http"
+		[ "${port}" == "8080" ] && service="http-alt"
+		[ "${port}" == "443" ] && service="https"
+		[ "${port}" == "8443" ] && service="pcsync-https"
+
+		procd_add_mdns_service "$service" "tcp" "$port" "daemon=nginx"
```

Comment:
Should service have a default value in case user does not use any of the ports you identify?
Or have:
```sh
case "${port}" in
    "80")
        service="http"
        ;;
    "8080")
        service="http-alt"
        ;;
    "443")
        service="https"
        ;;
    "8443")
        service="pcsync-https"
        ;;
    *)
        service="generic http"
        ;;
esac

```

Reply:
Right, updated.